### PR TITLE
Stream Output Error Handling Improvement

### PR DESCRIPTION
**Pull request by hiimjako**
## Pull Request Type

- [X] Bugfix

## Description

Currently, the stream output is validated according to the collection schema. However, if a document is not valid, the service crashes as there is no error handling in the output pipeline stream. This PR aims to address this issue by handling the error closing the stream (giving socketError), thereby preventing the CRUD service from crashing.

## PR Checklist

- [X] The commit message follows our guidelines included in the [CONTRIBUTING.md](../CONTRIBUTING.md#how-to-submit-a-pr)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -59,7 +63,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - `__mia_configuration` property in a collection `schema` now accepts additional properties
 - improved validation message for "body must NOT have additional properties" Ajv error, now it also says the unwanted property
-- optimization of the __STATE__ query sent to MongoDB: $in operator has been removed when not necessary
+- optimization of the **STATE** query sent to MongoDB: $in operator has been removed when not necessary
```
Comment (danibix95) [inline CHANGELOG.md:66]:
```suggestion
- optimization of the `__STATE__` query sent to MongoDB: $in operator has been removed when not necessary
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -210,9 +214,9 @@ This required to review source code and tests according to the [migration guide]
 
 ### Add
 
-- Add new _rawp's operators: $eq, $gt, $gte, $lt, $lte, $ne, $nin, $and, $not, $nor, $or, $exists, $type, $all, $elemMatch, $size, $cond, $regexMatch, $map, $mod
+- Add new \_rawp's operators: $eq, $gt, $gte, $lt, $lte, $ne, $nin, $and, $not, $nor, $or, $exists, $type, $all, $elemMatch, $size, $cond, $regexMatch, $map, $mod
 
-- Add the *CRUD_MAX_LIMIT* environment variable, for setting up the maximum limit of object per query
+- Add the _CRUD_MAX_LIMIT_ environment variable, for setting up the maximum limit of object per query
```
Comment (danibix95) [inline CHANGELOG.md:219]:
```suggestion
- Add the `CRUD_MAX_LIMIT` environment variable, for setting up the maximum limit of object per query
```

File: lib/httpInterface.js [hunk 3]
```diff
@@ -268,28 +269,29 @@ function handleGetListLookup(request, reply) {
...
+      this.crudService
+        .aggregate(crudContext, mongoQuery, projection, sort, skip, limit, stateArr, isTextSearchQuery)
+        .stream(),
+      this.castResultsAsStream(),
+      validateStream(validate),
+      serializer(),
+      reply.raw
+    )
+  } catch (error) {
+    request.log.error({ error }, 'Error during findAll lookup stream')
```
Comment (danibix95) [inline lib/httpInterface.js:284]: We need to consider whether this is working properly or not. In fact, the default pino Error serializer is configured with `err` property and not error 🤔

Maybe we can bypass it by just setting the error as the object to be logged:
```suggestion
    request.log.error(error, 'Error during findAll lookup stream')
```

File: lib/httpInterface.js [hunk 4]
```diff
@@ -268,28 +269,29 @@ function handleGetListLookup(request, reply) {
...
+    // NOTE: error from Mongo may not serialize the message, so we force it here,
+    // we use debug level to prevent leaking potetially sensible information in logs.
+    request.log.debug({ error: { ...error, message: error.message } }, 'Error during findAll lookup stream with message')
```
Comment (danibix95) [inline lib/httpInterface.js:287]: In case it is an error from Mongo, we may also log the `code` of the error

File: lib/httpInterface.js [hunk 5]
```diff
@@ -335,25 +337,25 @@ function handleGetList(request, reply) {
...
+      this.crudService
+        .findAll(crudContext, mongoQuery, projection, sort, skip, limit, stateArr, isTextSearchQuery)
+        .stream(),
+      this.castResultsAsStream(),
+      validateStream(validate),
+      serializer(),
+      reply.raw
+    )
+  } catch (error) {
+    request.log.error({ error }, 'Error during findAll stream')
```
Comment (danibix95) [inline lib/httpInterface.js:351]: Same as above

File: lib/httpInterface.js (hunk 4 above)
Comment (hiimjako) [reply lib/httpInterface.js:287]: It is already logged

File: lib/httpInterface.js (hunk 3 above)
Comment (hiimjako) [reply lib/httpInterface.js:284]: Errors are already verbose, they are logged as expected, so I do not think it is necessary to edit them.

**Pull request merged**